src/panels/lovelace/entity-rows/hui-media-player-entity-row.ts

```diff
@@ -0,0 +1,197 @@
+import { html, LitElement } from "@polymer/lit-element";
+import { TemplateResult } from "lit-html";
+import "@polymer/paper-icon-button/paper-icon-button";
+
+import "../components/hui-generic-entity-row";
+
+import { hassLocalizeLitMixin } from "../../../mixins/lit-localize-mixin";
+import { EntityRow, EntityConfig } from "./types";
+import { HomeAssistant } from "../../../types";
+import { HassEntity } from "home-assistant-js-websocket";
+
+const SUPPORT_PAUSE = 1;
+const SUPPORT_NEXT_TRACK = 32;
+const SUPPORTS_PLAY = 16384;
+const OFF_STATES = ["off", "idle"];
+
+class HuiMediaPlayerEntityRow extends hassLocalizeLitMixin(LitElement)
+  implements EntityRow {
+  public hass?: HomeAssistant;
+  private _config?: EntityConfig;
+
+  static get properties() {
+    return {
+      hass: {},
+      _config: {},
+    };
+  }
+
+  public setConfig(config: EntityConfig): void {
+    if (!config || !config.entity) {
+      throw new Error("Invalid Configuration: 'entity' required");
+    }
+
+    this._config = config;
+  }
+
+  protected render(): TemplateResult {
+    if (!this.hass || !this._config) {
+      return html``;
+    }
+
+    const stateObj = this.hass.states[this._config.entity];
+
+    if (!stateObj) {
+      return html`
+        <hui-error-entity-row
+          .entity="${this._config.entity}"
+        ></hui-error-entity-row>
+      `;
+    }
+
+    return html`
+      ${this.renderStyle()}
+      <hui-generic-entity-row
+        .hass="${this.hass}"
+        .config="${this._config}"
+        .showSecondary="false"
+      >
+        ${
+          !this._isOff(stateObj.state)
+            ? html`
+                <div class="controls">
+                  ${
+                    this._computeControlIcon(stateObj)
```

Comment:
This should just check if the state === playing. We should probably just remove this method.

Reply:
I don't follow here. We want to show controls regardless of state, it is just a matter of if the device supports the features on which icon we render.

Reply:
Well you're using the computing of the icon to see if you should render the icon. You should extract that check and just do that check, not generate the icon (just to check if you should render)